src/libertem/io/dataset/k2is.py

```diff
@@ -507,16 +513,40 @@ def __init__(self, path):
         # skip_frames is applied after synchronization.
         self._skip_frames = -1
         self._files = None
+        self._reshape = reshape
+        self._sync_offset = 0
+        self._sync_offset_info = None
+        if self._reshape and "sync_offset" in self._reshape:
+            self._sync_offset = self._reshape['sync_offset']
 
     def _do_initialize(self):
         self._files = self._get_files()
         self._get_syncer(do_sync=True)
         self._scan_size = self._get_scansize()
+        self._scan_size_product = int(np.prod(self._scan_size))
```

Comment:
So... K2IS is a bit of a special format in that the scan size we can read from the `gtg` file is not equal to the "real" number of frames in the .bin files. In the .bin files, each block of data has a small header, which has, among others, a `sync` flag. So in a typical K2IS data set, you get a few frames of data at the beginning where the sync flag is not set, and then you get the rest of the data, with the `sync` flag set.

So, `_get_scansize` will, basically, only give you the number of frames that have the `sync` flag set. The underlying data can have more frames before that, and possibly also after (I haven't checked that one, would be interesting to play around with).

In the current version, we use this `sync` flag to automatically find something like a `sync_offset` (see the `K2Syncer` class, `first_block_offset`, `_start_offsets` and `_skip_frames`), but in the future we should allow to access the frames without the `sync` flag set. So, in summary: `scan_size_product` != `image_count`.

Let me know if this is clear!

Reply:
Thanks! I'll try to get a better understanding of this and let you know later.

Reply:
Opened #911 and resolving this since we decided to solve this in another PR